std/stdio.d

```diff
@@ -3084,11 +3084,24 @@ void main()
         import std.exception : collectException;
 
         ulong pos = void;
-        if (collectException(pos = tell)) return ulong.max;
+        if (collectException(pos = tell))
+            return ulong.max;
+
+        // Unfortunately there doesn't seem to be a reliable way to
+        // find out if a file is seekable without actually attempting
+        // to seek in it. Even seek(0, SEEK_CUR) will succeed in
+        // unseekable files.
+        if (collectException(seek(0, SEEK_END)))
```

Comment:
I guess the `seek` in the `scope` can still throw.

It's a pretty ugly situation. I looked into better ways to detect seekability, but even errno is set to a very unhelpful "invalid argument".